format-common/src/main/java/io/cdap/plugin/format/input/PathTrackingConfig.java

```diff
@@ -138,18 +144,20 @@ public Path getFilePathForSchemaGeneration(String path, String regexPathFilter,
     if (files.length == 0) {
       throw new IllegalArgumentException("Provided directory is empty");
     }
-
+    Pattern pattern = Pattern.compile(regexPathFilter);
     for (FileStatus file : files) {
       if (Strings.isNullOrEmpty(regexPathFilter)) {
```

Comment:
If the filter string is null, the compilation of Pattern would fail. You should move this check to outside of the loop such that you can have `pattern` set to `null` if no filter is provided and then checks for `pattern == null` here.

Reply:
Thanks, finish checking the filter.